src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,185 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            firstAdd(value);
+        } else {
+            addAfter(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheck(index);
+        if (size == 0 && index == 0) {
+            firstAdd(value);
+        } else if (index == 0 && size > 0) {
+            addBefore(value);
+        } else if (index == size) {
+            addAfter(value);
+        } else {
+            addMiddle(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (size == index) {
+            throw new IndexOutOfBoundsException("Index out of bound");
+        }
+        indexCheck(index);
+        return startFrom(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        add(value, index);
+        return remove(index + 1);
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        if (index == 0) {
+            return removeFirst();
+        } else if (index == size - 1) {
+            return removeLast();
+        } else {
+            return removeMiddle(index);
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> x = first;
```

Comment:
```suggestion
        Node<T> currentNode = first;
```

`x` - not informative name